tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -1017,8 +1017,15 @@ def _split_on_axis(np_fun_name, axis):
   def f(ary, indices_or_sections):
     # for 1-D array, hsplit becomes vsplit
     nonlocal axis
-    if axis == 1 and len(ary.shape) == 1:
-      axis = 0
+    axis = np_utils.cond(
+      math_ops.equal(axis, 1),
+      lambda: np_utils.cond(
+        math_ops.equal(array_ops.rank(ary), 1),
+        lambda: 0,
+        lambda: 1
+      ),
+      lambda: 1
+    )
```

Comment:
For the test can you please add it to tensorflow/python/ops/numpy_ops/np_array_ops_test.py? Thanks